packages/core/src/application.ts

```diff
@@ -80,7 +81,8 @@ export class Application extends Context {
       name,
       namespace: CoreBindings.SERVERS,
       type: 'server',
-    }).inScope(BindingScope.SINGLETON);
+      defaultScope: BindingScope.SINGLETON,
```

Comment:
Please add a test to verify this change.

I think this is the scenario:
 - A server  class is configured with TRANSIENT scope
 - The class is added to an app via `app.server` API
 - The server binding is expected to have TRANSIENT scope